src/partitioner/local/RDFParser.cpp

```diff
@@ -0,0 +1,422 @@
+    // Can't call transcode till after Xerces Initialize()
+    TAG_root = XMLString::transcode("root");
+    TAG_RDF = XMLString::transcode("rdf:RDF");
+    TAG_has_author = XMLString::transcode("akt:has-author");
+
+    TAG_has_title = XMLString::transcode("akt:has-title");
+    TAG_article_of_journal = XMLString::transcode("akt:article-of-journal");
+    TAG_Journal = XMLString::transcode("akt:Journal");
+    TAG_has_volume = XMLString::transcode("akt:has-volume");
+    TAG_has_web_address = XMLString::transcode("akt:has-web-address");
+    TAG_Calendar_date = XMLString::transcode("akts:Calendar-Date");
+    TAG_year_of = XMLString::transcode("akts:year-of");
+    TAG_month_of = XMLString::transcode("akts:month-of");
+    TAG_has_date = XMLString::transcode("akt:has-date");
+
+
+    TAG_Article_Reference = XMLString::transcode("akt:Article-Reference");
+    TAG_Book_Section_Reference = XMLString::transcode("akt:Book-Section-Reference");
+    TAG_Thesis_Reference = XMLString::transcode("akt:Thesis-Reference");
+    TAG_Book_Reference = XMLString::transcode("akt:Book-Reference");
+    TAG_Conference_Proceedings_Reference = XMLString::transcode("akt:Conference-Proceedings-Reference");
+
+    TAG_Person = XMLString::transcode("akt:Person");
+    TAG_full_name = XMLString::transcode("akt:full-name");
+
+
+    m_ConfigFileParser = new XercesDOMParser;
+}
+
+/**
+ *  Class destructor frees memory used to hold the XML tag and
+ *  attribute definitions. It als terminates use of the xerces-C
+ *  framework.
+ */
+
+GetConfig::~GetConfig() {
+    // Free memory
+
+    delete m_ConfigFileParser;
+
+    try {
+        XMLString::release(&TAG_root);
+
+        XMLString::release(&TAG_RDF);
+        XMLString::release(&TAG_has_author);
+        XMLString::release(&TAG_has_title);
+        XMLString::release(&TAG_article_of_journal);
+        XMLString::release(&TAG_Journal);
+        XMLString::release(&TAG_has_volume);
+        XMLString::release(&TAG_has_web_address);
+        XMLString::release(&TAG_Calendar_date);
+        XMLString::release(&TAG_year_of);
+        XMLString::release(&TAG_month_of);
+        XMLString::release(&TAG_has_date);
+        XMLString::release(&TAG_Article_Reference);
+        XMLString::release(&TAG_Book_Section_Reference);
+        XMLString::release(&TAG_Thesis_Reference);
+        XMLString::release(&TAG_Book_Reference);
+        XMLString::release(&TAG_Conference_Proceedings_Reference);
+        XMLString::release(&TAG_Person);
+        XMLString::release(&TAG_full_name);
+    }
+    catch (...) {
+        cerr << "Unknown exception encountered in TagNamesdtor" << endl;
+    }
+
+    // Terminate Xerces
+
+    try {
+        XMLPlatformUtils::Terminate();  // Terminate after release of memory
+    }
+    catch (xercesc::XMLException &e) {
+        char *message = xercesc::XMLString::transcode(e.getMessage());
+
+        cerr << "XML ttolkit teardown error: " << message << endl;
```

Comment:
ttolkit?

Reply:
This is also taken from the sample code I referred. Should I change the error message?

Reply:
Yes, please change this to meaningful error message. The word "ttolkit" does not have any meaning. We should have meaningful error message.